SourceHgWeb/SourceHgWeb.php

```diff
@@ -268,7 +268,7 @@ private function commit_changeset( $p_repo, $p_input, $p_branch='' ) {
 
 			$t_changeset->author_email = empty($t_commit['author_email'])? '': $t_commit['author_email'];
 
-			preg_match_all('#diff[\s]*-r[\s]([^\s]*)[\s]*-r[\s]([^\s]*)[\s]([^\n]*)\n(Binary file[\s]([^\s]*)[\s]has changed|\-{3}[\s](/dev/null)?[^\t]*[^\n]*\n\+{3}[\s](/dev/null)?[^\t]*\t[^\n]*)#', $p_input, $t_matches, PREG_SET_ORDER);
+			preg_match_all('#diff[\s]*-r[\s]([^\s]*)[\s]*-r[\s]([^\s]*)[\s]([^\n]*)\n(Binary file[\s]([^\n]*)[\s]has changed|\-{3}[\s](/dev/null)?[^\t]*[^\n]*\n\+{3}[\s](/dev/null)?[^\t]*\t[^\n]*)#', $p_input, $t_matches, PREG_SET_ORDER);
```

Comment:
Sorry for the late response and thank you for point out my fault. Because I have to match Chinese characters, but I was too greedy by using `([^\n]*)` only. According to [PHP page](http://php.net/manual/en/regexp.reference.unicode.php), I will change it to `([/P{C}]*)` and have a test.